clients/rospy/src/rospy/core.py

```diff
@@ -145,44 +145,42 @@ def rospywarn(msg, *args):
     _rospy_logger.warn(msg, *args)
     
 
-def _base_logger(msg, *args, **kwargs):
-
-    name = kwargs.pop('logger_name', None)
-    throttle = kwargs.pop('logger_throttle', None)
-    level = kwargs.pop('logger_level', None)
-    once = kwargs.pop('logger_once', False)
+def _base_logger(msg, args, kwargs, throttle=None,
+                 level=None, once=False):
 
     rospy_logger = logging.getLogger('rosout')
+    name = kwargs.get('logger_name')
     if name:
         rospy_logger = rospy_logger.getChild(name)
+        del kwargs['logger_name']
```

Comment:
Instead of `kwargs.get(...)` and `del kwargs[...]` could this continue to call `kwargs.pop(...)` as it was before?